server/views/components/series-container/index.njk

```diff
@@ -12,26 +12,42 @@
   </div>
 </div>
 
-<div class="row row--small-top-padding">
-  <div class="container">
-    <div class="grid">
-      {% for promo in model.items.take(3).toJS() %}
-        <div class="{{ {s:4, l:4, xl:4} | gridClasses }}">
-          {% component 'promo', { promo: promo } %}
-        </div>
-      {% endfor %}
+{% set themeSize = model.items.size %}
+
+{% if themeSize <= 6 %}
+  <div class="row row--small-top-padding">
+    <div class="container">
+      <div class="grid grid--theme-{{ themeSize }}">
+        {% for promo in model.items.take(6).toJS() %}
+          <div class="grid__cell">
+            {% component 'promo', { promo: promo } %}
+          </div>
+        {% endfor %}
+      </div>
+    </div>
+  </div>
+{% else %}
+  <div class="row row--small-top-padding">
+    <div class="container">
+      <div class="grid grid--theme-many">
+        {% for promo in model.items.take(3).toJS() %}
+          <div class="grid__cell">
+            {% component 'promo', { promo: promo } %}
+          </div>
+        {% endfor %}
+      </div>
     </div>
   </div>
-</div>
 
-<div class="row row--no-padding">
-  <div class="container container--scroll touch-scroll">
-    <div class="grid grid--featured">
-      {% for promo in model.items.skip(3).toJS() %}
-        <div class="{{ {s:4, m:4, l:3, xl:3} | gridClasses }}">
-          {% component 'promo', { promo: promo } %}
-        </div>
-      {% endfor %}
+  <div class="row row--no-padding">
+    <div class="container container--scroll container--no-max-width touch-scroll">
+      <div class="grid grid--scroll">
+        {% for promo in model.items.skip(3).toJS() %}
+          <div class="{{ {s:4, m:4, l:3, xl:3} | gridClasses }}">
```

Comment:
Do we still need the sizes in this case, aren't they now determined by the grid--theme-x styles?